proposals/bulk-memory-operations/Overview.md

```diff
@@ -171,93 +169,109 @@ Since WebAssembly currently does not allow for multiple memories, the memory
 `memory.init` followed by a `data.drop` (or `table.init` followed by a
 `elem.drop`) that is prepended to the module's start function.
 
-The new encoding of a data segment is now:
+Additionally, the reference-types proposal introduces the notion of a function
+reference (a function whose address is a program value).  To support this,
+element segments can have several encodings, and can also be used to
+forward-declare functions whose address will be taken; see below.
+
+The reference-types proposal also introduces the bulk instructions `table.fill`
+and `table.grow`, both of which take a function reference as an initializer
+argument.
+
+### Data segments
+
+The meaning of the bits of the flag field for data segments is:
+
+| Bit | Meaning                                        |
+| -   | -                                              |
+| 0   | 0=is active, 1=is passive                      |
+| 1   | if bit 0 clear: 0=memory 0, 1=has memory index |
 
-| Field | Type | Present? | Description |
-| - | - | - | - |
-| flags | `varuint32` | always | Flags for passive and presence of fields below, only values of 0, 1, and 2 are valid |
-| index | `varuint32`? | flags = 2 | Memory index; 0 if the field is not present |
-| offset | `init_expr`? | flags != 1 | an `i32` initializer expression for offset |
-| size | `varuint32` | always | size of `data` (in bytes) |
-| data | `bytes` | always | sequence of `size` bytes |
+which yields this view, with the fields carried by each flag value:
 
-Another way of looking at it:
+| Flags | Meaning                  | Memory index | Offset in memory | Count       | Payload |
+| -     | -                        | -            | -                | -           | -       |
+| 0     | Active                   |              | `init_expr`      | `varuint32` | `u8`*   |
+| 1     | Passive                  |              |                  | `varuint32` | `u8`*   |
+| 2     | Active with memory index | `varuint32`  | `init_expr`      | `varuint32` | `u8`*   |
+
+All other flag values are illegal.  At present the memory index must be zero,
+but the upcoming multi-memory proposal changes that.
 
-| Flags | Active? | index | offset |
-| - | - | - | - |
-| 0 | Active | Always 0 | Present |
-| 1 | Passive | - | -  |
-| 2 | Active | Present | Present |
 
 ### Element segments
 
-The new binary format for element segments is similar to the new format for data segments, but
-also includes an element type when the segment is passive. A passive segment also has a sequence
-of `expr`s instead of function indices.
+The meaning of the bits of the flag field for element segments is:
+
+| Bit | Meaning                                          |
+| -   | -                                                |
+| 0   | 0=is active, 1=is passive                        |
+| 1   | if bit 0 clear: 0=table 0, 1=has table index     | 
+|     | if bit 0 set: 0=active, 1=declared               |
```

Comment:
You may want to add a note regarding `declared` since that actually isn't part of this proposal.